src/cgutils.cpp

```diff
@@ -3282,7 +3296,11 @@ static void recursively_adjust_ptr_type(llvm::Value *Val, unsigned FromAS, unsig
     for (auto *User : Val->users()) {
         if (isa<GetElementPtrInst>(User)) {
             GetElementPtrInst *Inst = cast<GetElementPtrInst>(User);
+            #if JL_LLVM_VERSION >= 170000
+            Inst->mutateType(PointerType::get(Inst->getType(), ToAS));
```

Comment:
@gbaraldi I believe this is the root of the assertion crash. The erroring assert is in https://github.com/llvm/llvm-project/blob/6009708b4367171ccdbf4b5905cb6a803753fe18/llvm/lib/IR/Constants.cpp#L1484, which processes a `GetElementPtr` instruction.